lib/puppet/agent.rb

```diff
@@ -84,6 +94,8 @@ def run(client_options = {})
           rescue StandardError => detail
             Puppet.log_exception(detail, _("Could not run %{client_class}: %{detail}") % { client_class: client_class, detail: detail })
             nil
+          ensure
+            Puppet.runtime[:http].close
```

Comment:
Trying to wrap my head around why this was added, is it to reset the settings for `default_system_ssl_context` and `default_system_ssl_context`?

Reply:
Yeah, it's to make sure that state is cleared across agent runs.